### PR TITLE
Diagrams

### PR DESCRIPTION
**Pull request by mbh226**
### **Diagrams:**

This pull request adds some initial UML dependency and sequence diagrams that were created using PlantUML. To view the diagrams in VSCode, you'll need to install the PlantUML extension in VSCode to render them. After install:

1. Open file
2. Right click anywhere in the code
3. Click "Preview Current Diagram

You will also need to have java and graphviz installed to view these diagrams using the PlantUML extension in VSCode. You can also find free PlantUML renderers online.

---

### **Description:**

This PR adds:

1. **`/diagrams/PriceApp_JSONValidation.puml`**:
   - This is a dependency diagram that shows how PriceApp.price uses JSON_Validation.validator and external libraries to perform our main functions.

2. **`Security_Scanning.puml`**:
   - This is a UML diagram that illustrates our security scanning workflow.
   - It includes details about the `convert_safety_to_sarif.py` script that is used to convert Safety's JSON output to SARIF format for uploading to GitHub Security Code Scanning alerts.

3. **`sequence_diagram.puml`**:
   - This is a sequence diagram that illustrates all function calls when running our main `calculate_value_sharpe()` function in `pricy.py`. 

6. **`stock-shema-json.puml`**:
   - This is a UML diagram that illustrates our agreed upon JSON portfolio schema.

7. **`UnitTests.puml`**:
   - This is a dependency diagram that illustrates our UnitTests, their relationship to `price.py`, and their required external libraries.

8. **`UnitTests_IncludingFuzz.puml`**:
   - This is a dependency diagram that illustrates our UnitTests, their relationship to `price.py`, and their required external libraries.
   - It includes Sandro's latest `test_fuzz.py` test.

This PR modifies:
1. **`requirements.txt`**:
- I updated the `requirements.txt` file to reflect our most up to date requirements.

2. **`price.py`**:
   - Removed import from `json` and `os` external libraries because we don't use them.
   - Removed very long, unnecessary comment about Sharpe ratio.
   - Removed a commented line of code indicating a function called `load_portfolio()` wasn't needed anymore.
   - Instead of just calling `calculate_value_sharpe()` at the end of the file, I added a main function. This helps with control flow and using the script as both a standalone program and an importable module.

### CODE REVIEW

File: PriceApp/price.py [hunk 1]
```diff
@@ -1,38 +1,9 @@
 import yfinance as yf
 from datetime import datetime
-import json
 import pandas as pd
-import os
-
 from JSON_Validation.validator import validate_portfolio, load_schema, load_portfolio
 
 def fetch_portfolio_sharpe_ratio(portfolio, price_data, total_investment, risk_free_rate=0.03):
-    """
```
Comment (kgp33) [inline PriceApp/price.py:12]: This is good to have because it gives a visual of what the data frame contains, so it would be good to keep this in

**Comment by kgp33** [comment]
This looks great. Only two things I would suggest changing is not deleting the data frame comment (I left a review). Additionally, if you can add in the read me the how to open the diagram, it will be visible after the merge:

"""Diagrams:
This pull request adds some initial UML dependency and sequence diagrams that were created using PlantUML. To view the diagrams in VSCode, you'll need to install the PlantUML extension in VSCode to render them. After install:

Open file
Right click anywhere in the code
Click "Preview Current Diagram
"""

**Review by kgp33 — COMMENTED** [review]
Can you re-add the description of the Data Frame. It is good to have because programmers can have a better idea of what it looks like when implementing new features.

File: PriceApp/price.py (hunk 1 above)
Comment (mbh226) [reply PriceApp/price.py:12]: I added it back in

Comment (mintfish3r) [reply PriceApp/price.py:12]: Looks good to me

**Review by mintfish3r — APPROVED** [review]
Everything looks good. Ready to merge.

**Pull request merged**